src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // Wraps a record
+    // This also preserves the lazy field enumeration semantics of a recordType.
+    internal class SymbolTableOverRecordType : ReadOnlySymbolTable, IGlobalSymbolNameResolver
+    {
+        private readonly RecordType _type;
+
+        public SymbolTableOverRecordType(RecordType type)
+        {
+            _type = type;
+            _debugName = "per-eval";
+        }
+
+        // Key is the logical name. 
+        // Display names are in the NameLookupInfo.DisplayName field.
+        IReadOnlyDictionary<string, NameLookupInfo> IGlobalSymbolNameResolver.GlobalSymbols
+        {
+            get
+            {
+                var map = new Dictionary<string, NameLookupInfo>();
+                foreach (var kv in _type.GetFieldTypes())
+                {
+                    map[kv.Name] = Create(kv.Name, kv.Type);
+                }
+
+                return map;
+            }
+        }
+
+        internal override bool TryLookup(DName name, out NameLookupInfo nameInfo)
+        {
+            // Lookup must handle display names
+            // This is consistent with RowScope. 
+            if (!DType.TryGetConvertedDisplayNameAndLogicalNameForColumn(_type._type, name, out var logicalName, out _))
```

Comment:
>out _ [](http://example.com/codeflow?start=113&length=5)

Note that this is safe to discard, as non-canvas PowerFx never attempts a display -> display rewrite.